scapy/packet.py

```diff
@@ -1844,40 +1851,97 @@ def explore(layer=None):
         # 2 - Retrieve list of Packets
         if action == "layers":
             # Get all loaded layers
-            _radio_values = conf.layers.layers()
+            values = conf.layers.layers()
             # Restrict to layers-only (not contribs) + packet.py and asn1*.py
-            _radio_values = [x for x in _radio_values if ("layers" in x[0] or
-                                                          "packet" in x[0] or
-                                                          "asn1" in x[0])]
+            values = [x for x in values if ("layers" in x[0] or
+                                            "packet" in x[0] or
+                                            "asn1" in x[0])]
         elif action == "contribs":
             # Get all existing contribs
             from scapy.main import list_contrib
-            _radio_values = list_contrib(ret=True)
-            _radio_values = [(x['name'], x['description'])
-                             for x in _radio_values]
+            values = list_contrib(ret=True)
+            values = [(x['name'], x['description'])
+                      for x in values]
             # Remove very specific modules
-            _radio_values = [x for x in _radio_values if not ("can" in x[0])]
+            values = [x for x in values if "can" not in x[0]]
         else:
             # Escape/Cancel was pressed
             return
         # Python 2 compat
         if six.PY2:
-            _radio_values = [(six.text_type(x), six.text_type(y))
-                             for x, y in _radio_values]
+            values = [(six.text_type(x), six.text_type(y))
+                      for x, y in values]
+        # Build tree
+        if action == "contribs":
+            tree = defaultdict(list)
+            for name, desc in values:
+                if "." in name:  # Folder detected
+                    parts = name.split(".")
+                    q = tree
```

Comment:
Could you change the one and two letters variables names to something more meaningful?

Reply:
Done. I also used `git mv` to have a commit move the file properly

Reply:
Thanks! these comments are useful =)